server/helper/ImageUtils.js

```diff
@@ -45,63 +88,69 @@ async function resizeImage(sourcePath, maxDimensions, targetPath) {
       });
   });
 
-  if (!replaceOriginal) return targetResult;
+  if (!replaceOriginal) return resizedImage;
 
-  await fs.rename(target, sourcePath, (err) => {
+  fs.rename(target, sourcePath, (err) => {
     if (err) logger.error("IU: " + err);
   });
 }
 
 /**
- * Creates a path for a thumbnail. The path of the base image will be extended by a postfix.
- * The position of a possible fileextension at the end of a path will be keeped.
- * @param {*} basePath
- * @returns
+ * Creates a path for a smaller size version of the base image. The path of the base image will be extended by a postfix.
+ * The position of a possible fileextension at the end of a path will be preserved.
+ *
+ * @param {String} basePath The path to the original image.
+ * @returns {String} A file name for a smaller size version of the original image.
  */
-function createThumbnailPath(basePath) {
+function createSizePath(basePath, size) {
   const pathAsString = basePath.toString();
+  const postfix = size ? "-" + size : "";
```

Comment:
Aktuell wird das nur von "retrieveFilePath" aufgerufen. D.h. size ist in createSizePath immer definiert.
Deshalb könnte man sich dann hier dann die Zeile 107 sparen.